Epilepsy/epilepsy_utils.py

```diff
@@ -0,0 +1,62 @@
+import cPickle
+
+from collections import defaultdict
+from itertools import product
+from pandas import DataFrame
+from string import punctuation
+
+#Ripped off from the snorkel custom tagger.
+#Designed to get the offset where a given token is found so it can receive a custom entity tag (i.e. Gene or Chemical)
+def offsets_to_token(left, right, offset_array, lemmas, punc=set(punctuation)):
+    token_start, token_end = None, None
+    for i, c in enumerate(offset_array):
+        if left >= c:
+            token_start = i
+        if c > right and token_end is None:
+            token_end = i
+            break
+    token_end = len(offset_array) - 1 if token_end is None else token_end
+    token_end = token_end - 1 if lemmas[token_end - 1] in punc else token_end
+    return range(token_start, token_end)
+
+
+
+#This is a custom class that is designed to tag each relevant word in a given sentence
+# i.e if it see GAD then this tagger will give it a Gene tag.
+class Tagger(object):
+    def __init__(self,file_name):
+        self.open_file = open(file_name,"r")
```

Comment:
Does this file ever get closed? In practice I don't think it's a huge deal to leave files open, but it's ideal to close them when done.

Since you're opening the file in __init__, it's not entirely clear where the file should get closed.